src/Model/Resolver/EntityUrl.php

```diff
@@ -141,10 +185,22 @@ public function resolve(
                 $product = $collection->addIdFilter($id)->getFirstItem();
                 $isInStock = false;
 
-                try {
-                    $isInStock = $this->stockItemRepository->get($id)->getIsInStock();
-                } catch (NoSuchEntityException $e) {
-                    // Ignoring error is safe
+                if ($this->isProductTypeBundle($product)) {
+                    $productType = 'bundle';
+                } else {
+                    $productType = $this->getProductType($id);
+                }
+
+                if ($productType === 'configurable') {
+                    $isInStock = $this->getConfigurableProductStockState($product);
+                } elseif ($productType === 'bundle') {
+                    $isInStock = $this->getBundleProductsStockState($this->getBundleProductChildrenItems($product, $id));
+                } else {
+                    try {
+                        $isInStock = $this->stockItemRepository->get($id)->getIsInStock();
+                    } catch (NoSuchEntityException $e) {
+                        // Ignoring error is safe
+                    }
                 }
 
```

Comment:
Now this is dirty, can you please try writing that in some less **iffy** way? switch? Maybe also move the product type obtaining into separate function?